eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -698,6 +708,155 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
       }
 
+    case Event(cmd: CMD_SPLICE, d: DATA_NORMAL) =>
+      val replyTo = if (cmd.replyTo == ActorRef.noSender) sender() else cmd.replyTo
+      d.spliceStatus match {
+        case SpliceStatus.NoSplice =>
+          val parentCommitments = d.metaCommitments.main
+          if (parentCommitments.hasNoPendingHtlcsOrFeeUpdate && parentCommitments.channelFeatures.hasFeature(DualFunding)) {
+            log.info(s"initiating splice with local.in.amount=${cmd.additionalLocalFunding} local.in.push=${cmd.pushAmount} out.amount=${cmd.spliceOut_opt.map(_.amount).sum}")
+            val txInitRbf = TxInitRbf(d.channelId,
+              lockTime = nodeParams.currentBlockHeight.toLong,
+              feerate = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.fundingBlockTarget),
+              fundingContribution = cmd.additionalLocalFunding,
+              pushAmount = cmd.pushAmount,
+              spliceOut_opt = cmd.spliceOut_opt)
+            stay() using d.copy(spliceStatus = SpliceStatus.SpliceRequested(cmd.copy(replyTo = replyTo), txInitRbf)) sending txInitRbf
+          } else {
+            log.warning("cannot do splice")
+            replyTo ! Status.Failure(CommandUnavailableInThisState(d.channelId, "CMD_SPLICE_IN", NORMAL))
+            stay()
+          }
+        case _ =>
+          log.warning("cannot initiate splice, another one is already in progress")
+          replyTo ! Status.Failure(InvalidRbfAlreadyInProgress(d.channelId))
+          stay()
+      }
+
+    case Event(msg: TxInitRbf, d: DATA_NORMAL) =>
+      d.spliceStatus match {
+        case SpliceStatus.NoSplice =>
+          val parentCommitments = d.metaCommitments.main
+          if (parentCommitments.hasNoPendingHtlcsOrFeeUpdate && parentCommitments.channelFeatures.hasFeature(DualFunding)) {
+            log.info(s"accepting splice with remote.in.amount=${msg.fundingContribution} remote.in.push=${msg.pushAmount}")
+            val txAck = TxAckRbf(d.channelId, fundingContribution = 0.sat) // only remote contributes to the splice
+            val fundingParams = InteractiveTxParams(
+              channelId = d.channelId,
+              isInitiator = d.metaCommitments.main.localParams.isInitiator,
+              localAmount = txAck.fundingContribution,
+              remoteAmount = msg.fundingContribution,
+              commonInput_opt = Some(parentCommitments.commitment.commitInput),
+              fundingPubkeyScript = d.metaCommitments.main.commitInput.txOut.publicKeyScript, // same pubkey script as before
+              lockTime = nodeParams.currentBlockHeight.toLong,
+              dustLimit = d.metaCommitments.main.localParams.dustLimit.max(d.metaCommitments.main.remoteParams.dustLimit),
+              targetFeerate = msg.feerate,
+              requireConfirmedInputs = RequireConfirmedInputs(forLocal = false, forRemote = false), // TODO: revisit this
```

Comment:
What do you mean? That when using zero-conf we need to avoid checking the previous funding input, because that one will potentially be unconfirmed, even if we require the *other* inputs to be confirmed?

Reply:
I didn't even think about chaining splices, but that's a very interesting point. I only thought about the scenario where one side trusts the other, but not the other way around (e.g. with Phoenix), then depending on the particular contributions we had to do something smart.

I think we it's actually not complicated. As the ACINQ node we should always use:
```suggestion
              requireConfirmedInputs = RequireConfirmedInputs(forLocal = false, forRemote = true), // TODO: revisit this
```
And exclude the common input from this check (which is already the case). WDYT?

Reply:
I agree, that would make sense!

Reply:
Okay, I'm still leaving it as-is for now, can't hardcode a non-symmetrical behavior or it will make tests impossible.